documentation/Generate-code-from-TypeSpec.md

```diff
@@ -0,0 +1,116 @@
+# Before you Start
+
+[TypeScript Azure SDK Design Guidelines](https://azure.github.io/azure-sdk/typescript_introduction.html) is the overall design guideline of the client SDK.
+
+## Prerequisites
+
+- [Node.js 18.x LTS](https://nodejs.org/en/download) or later
+- [Git](https://git-scm.com/downloads)
+- [azure-rest-api-specs repo](https://github.com/Azure/azure-rest-api-specs)
+- [azure-sdk-for-js repo](https://github.com/Azure/azure-sdk-for-js/)
+
+# Generate SDK
+
+## Use TypeSpec defined in REST API specifications
+
+It is recommended to configure TypeSpec package on [REST API specifications](https://github.com/Azure/azure-rest-api-specs). Please refer to [these guidelines](https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/typespec-structure-guidelines.md).
+
+### How to configure tspconfig.yaml
+You can reference these two config files to configure the Modular or RLC package:
+- [Modular tspconfig.yaml](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml)
+- [RLC tspconfig.yaml](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/ai/Face/tspconfig.yaml)
+
+Please make sure `service-dir`, `package-dir`, `packageDetails`, `isModularLibrary`, `generateMetadata`, `flavor`(for typespec-ts) is correctly configured. `experimentalExtensibleEnums`, `enableOperationGroup`, `hierarchyClient` are the optional configs.
+If you want to enable sample generation with typespec-ts, you should add 
+```
+generateSample:true
+```
+in your tspconfig.yaml
+
+
+- "parameters.service-dir.default" would be `sdk/<service>`
+- "options.@azure-tools/typespec-ts.package-dir" would be `<module>`
+
+SDK module would be generated under the SDK project folder at `sdk/<service>/<module>`.
+
+### Generate Code with code-gen-pipeline tool (recommend)
+**Notice** These steps are to generate code using the local spec repo. If you want to generate code with the github url, please refer [Generate Code with tsp-client tool](#Generate-Code-with-tsp-client-tool)
+
+Install dependencies to use code-gen-pipeline,  
+```ps
+npm install -g @azure-tools/typespec-client-generator-cli@0.14.1
+npm install -g @microsoft/rush@5.92.0
+npm install -g @azure-tools/js-sdk-release-tools
+```
+
+Create a local json file named generatedInput.json with content similar to that shown below
+```
+  {
+    "dryRun": false,
+    "specFolder": <your-local-spec-repo-path>,
+    "headSha": <commit-id-you-want-to-generate-from>,
+    "repoHttpsUrl": "https://github.com/Azure/azure-rest-api-specs",
```

Comment:
Do we support the private repo or other fork of the spec repo?

Reply:
This is only used for updating `tsp-location.yaml`. So it can be any repo. 
e.g. https://github.com/Azure/azure-sdk-for-js/blob/b05f9966c17e350548eedeaa6512d98309b8eb34/sdk/openai/openai/tsp-location.yaml#L3
Pleae correct me if I'm wrong. @raych1 .